src/wallets/WalletContractV5.ts

```diff
@@ -0,0 +1,310 @@
+        await this.send(provider, transfer);
+    }
+
+    /**
+     * Sign and send add extension request
+     */
+    async sendAddExtension(provider: ContractProvider, args: Wallet5SendArgs & { extensionAddress: Address }) {
+        const request = this.createAddExtension(args);
+        await this.send(provider, request);
+    }
+
+    /**
+     * Sign and send remove extension request
+     */
+    async sendRemoveExtension(provider: ContractProvider, args: Wallet5SendArgs & { extensionAddress: Address, }) {
+        const request = this.createRemoveExtension(args);
+        await this.send(provider, request);
+    }
+
+    /**
+     * Sign and send request
+     */
+    async sendRequest(provider: ContractProvider, args: Wallet5SendArgs & { actions: (OutActionSendMsg | OutActionExtended)[], }) {
+        const request = this.createRequest(args);
+        await this.send(provider, request);
+    }
+
+    private createActions( args: {  messages: MessageRelaxed[], sendMode?: Maybe<SendMode> }) {
+        const sendMode = args.sendMode ?? SendMode.PAY_GAS_SEPARATELY + SendMode.IGNORE_ERRORS;
+        const actions: OutActionSendMsg[] = args.messages.map(message => ({ type: 'sendMsg', mode: sendMode, outMsg: message}));
+        return actions;
+    }
+
+    /**
+     * Create signed transfer
+     */
+    createTransfer(args: Wallet5SendArgs & { messages: MessageRelaxed[] }) {
+        const { messages, ...rest } = args;
+        return this.createRequest({
+            ...rest,
+            actions: this.createActions({ messages, sendMode: args.sendMode })
+        })
+    }
+
+    /**
+     * Create signed transfer async
+     */
+    createTransferAndSignRequestAsync(args: ExternallySingedAuthWallet5SendArgs & { messages: MessageRelaxed[] }) {
+        const { messages, ...rest } = args;
+        return this.createAndSignRequestAsync({
+            ...rest,
+            actions: this.createActions({ messages, sendMode: args.sendMode })
+        })
+    }
+
+    /**
+     * Create signed add extension request
+     */
+    createAddExtension(args: Wallet5SendArgs & { extensionAddress: Address }) {
+        const { extensionAddress, ...rest } = args;
+        return this.createRequest({
+            actions: [{
+                type: 'addExtension',
+                address: extensionAddress
+            }],
+            ...rest
+        })
+    }
+
+    /**
+     * Create signed remove extension request
+     */
+    createRemoveExtension(args: Wallet5SendArgs & { extensionAddress: Address }) {
+        const { extensionAddress, ...rest } = args;
+        return this.createRequest({
+            actions: [{
+                type: 'removeExtension',
+                address: extensionAddress
+            }],
+            ...rest
+        })
+    }
+
+    /**
+     * Create signed request or extension auth request
+     */
+    createRequest(args: Wallet5SendArgs & { actions: (OutActionSendMsg | OutActionExtended)[] }) {
```

Comment:
Why do not create specific type for `OutActions` allowed for WalletV5? To avoid using union

Reply:
fixed